lta/bundler.py

```diff
@@ -0,0 +1,324 @@
+            self.logger.info(f"Writing {num_files} files to bundle '{bundle_file_path}'")
+            file_count = 1
+            for bundle_me in results:
+                bundle_me_path = bundle_me["catalog"]["logical_name"]
+                self.logger.info(f"Writing file {file_count}/{num_files}: '{bundle_me_path}' to bundle '{bundle_file_path}'")
+                bundle_zip.write(bundle_me_path, os.path.basename(bundle_me_path))
+                file_count = file_count + 1
+        # 3. Put bundle in staging storage
+        final_bundle_path = bundle_file_path
+        if self.outbox_path != self.workbox_path:
+            final_bundle_path = os.path.join(self.outbox_path, f"{bundle_id}.zip")
+            self.logger.info(f"Moving bundle from '{bundle_file_path}' to '{final_bundle_path}'")
+            shutil.move(bundle_file_path, final_bundle_path)
+        self.logger.info(f"Finished archive bundle now located at: '{final_bundle_path}'")
+        # 3.1. Compute the SHA512 checksum of the bundle
+        self.logger.info(f"Computing checksum for bundle: '{final_bundle_path}'")
+        checksum = sha512sum(final_bundle_path)
+        self.logger.info(f"Bundle '{final_bundle_path}' has checksum '{checksum}'")
+        # 3.2. Clean up generated JSON metadata file
+        self.logger.info(f"Deleting bundle metadata file: '{metadata_file_path}'")
+        os.remove(metadata_file_path)
+        self.logger.info(f"Bundle metadata '{metadata_file_path}' was deleted.")
+        # 4. Add bundle to REST
+        bundle_obj = {
+            "bundles": [
+                {
+                    "bundle_uuid": bundle_id,
+                    "location": f"{source}:{final_bundle_path}",
+                    "checksum": {
+                        "sha512": checksum,
+                    },
+                    "status": "none",
+                    "verified": False,
+                    "manifest": [x["catalog"] for x in results],
+                },
+            ],
+        }
+        self.logger.info(f"POST /Bundles/actions/bulk_create - '{bundle_obj}'")
+        await lta_rc.request('POST', '/Bundles/actions/bulk_create', bundle_obj)
+        # 4.1. Update the files as bundled
+        self.logger.info(f"Marking Files as complete in the LTA DB")
+        update_body = {
+            "update": {
+                "complete": {
+                    "timestamp": now(),
+                    "bundler": self.bundler_name,
+                    "bundle_uuid": bundle_id,
+                }
+            },
+            "files": [x["catalog"]["uuid"] for x in results]
+        }
+        await lta_rc.request('POST', '/Files/actions/bulk_update', update_body)
+        # 4.2. Remove the files from the DB
+        delete_body = {
+            "files": [x["catalog"]["uuid"] for x in results]
+        }
+        await lta_rc.request('POST', f'/Files/actions/bulk_delete', delete_body)
```

Comment:
Why are we deleting the files here?  What if the files go to two different destinations - won't this cause the second destination to miss the files?

I thought the `deleter` module was supposed to delete files, after they were verified to be at the destinations.

Reply:
Got rid of this code block, so no more deleting.